web_poet/overrides.py

```diff
@@ -0,0 +1,146 @@
+import importlib
+import importlib.util
+import pkgutil
+import sys
+from dataclasses import dataclass, field
+from typing import Iterable, Union, List, Callable, Dict, Any
+
+from url_matcher import Patterns
+
+
+OVERRIDES_NAMESPACES_KEY = "_overrides_namespaces_"
+
+
+@dataclass(frozen=True)
+class HandleUrlsSpec:
+    """Meta information used by the :py:func:`web_poet.handle_urls` decorator"""
+    patterns: Patterns
+    overrides: Callable
+    meta: Dict[str, Any] = field(default_factory=dict)
+
+
+@dataclass(frozen=True)
+class OverrideRule:
+    """A single override rule. Specify when a page object should be used instead of another"""
+    for_patterns: Patterns
+    use: Callable
+    instead_of: Callable
+    meta: Dict[str, Any] = field(default_factory=dict)
+
+
+def _as_list(value: Union[str, Iterable[str], None]) -> List[str]:
+    """
+    >>> _as_list(None)
+    []
+    >>> _as_list("foo")
+    ['foo']
+    >>> _as_list(["foo", "bar"])
+    ['foo', 'bar']
+    """
+    if value is None:
+        return []
+    if isinstance(value, str):
+        return [value]
+    return list(value)
+
+
+def handle_urls(include: Union[str, Iterable[str]],
+                overrides: Callable,
+                *,
+                exclude: Union[str, Iterable[str], None] = None,
+                priority: int = 500,
+                namespace: str = "",
```

Comment:
I would suggest adding `namespace` description to the docstring below. It looks like the only missing one.

Reply:
The namespace approach was replaced by using Classes instead as suggested by Mikhail. Refactored in https://github.com/scrapinghub/web-poet/pull/16/commits/f5cffefd521e17c419ef2db004d0fbc717417aa8.